src/plugins/es_ui_shared/docs/docs/form_lib/examples/react_to_changes.md

```diff
@@ -0,0 +1,189 @@
+  const { form } = useForm();
+  const [{ showAddress }] = useFormData({ form, watch: 'showAddress' });
+
+  return (
+    <Form form={form}>
+      {/* Changing the "name" field won't trigger a re-render */}
+      <UseField path="name" config={{ label: 'First name' }} component={TextField} />
+
+      <UseField
+        path="showAddress"
+        config={{ defaultValue: false, label: 'Show address' }}
+        component={ToggleField}
+      />
+      {showAddress && (
+        <>
+          <p>800 W El Camino Real #350</p>
+        </>
+      )}
+    </Form>
+  );
+};
+```
+
+## Using the `onChange` handler
+
+Sometimes the good old `onChange` handler is all you need to react to a form field value change (instead of reading the form data and add a `useEffect` to react to it).
+
+```js
+export const OnChangeHandler = () => {
+  const { form } = useForm();
+
+  const onNameChange = (value: string) => {
+    console.log(value);
+  };
+
+  return (
+    <Form form={form}>
+      <UseField
+        path="name"
+        config={{ label: 'Name' }}
+        component={TextField}
+        onChange={onNameChange}
+      />
+    </Form>
+  );
+};
+```
+
+## Forward the form state to a parent component
+
+If your UX requires to submit the form in a parent component (e.g. because that's where your submit button is located), you will need a way to access the form validity and the form data outside your form component. Unless your parent component needs to be aware of every field value change in the form (which should rarely be needed), you don't want to use the `useFormData()` hook and forward the data from there. This would create unnecessary re-renders. Instead it is better to forward the `getFormData()` handler on the form.
+
+```js
+interface MyForm {
+  name: string;
+}
+
+interface FormState {
+  isValid: boolean | undefined;
+  validate(): Promise<boolean>;
+  getData(): MyForm;
+}
+
+const schema: FormSchema<MyForm> = {
+  name: {
+    validations: [
+      {
+        validator: ({ value }) => {
+          if (value === 'John') {
+            return { message: `The username "John" already exists` };
+          }
+        },
+      },
+    ],
+  },
+};
+
+interface Props {
+  defaultValue: MyForm;
+  onChange(formState: FormState): void;
+}
+
+const MyForm = ({ defaultValue, onChange }: Props) => {
+  const { form } = useForm<MyForm>({ defaultValue, schema });
+  const { isValid, validate, getFormData } = form;
+
+  useEffect(() => {
+    onChange({ isValid, validate, getData: getFormData });
+  }, [onChange, isValid, validate, getFormData]);
+
+  return (
+    <Form form={form}>
+      <UseField path="name" component={TextField} />
+    </Form>
+  );
+};
+
+export const ForwardFormStateToParent = () => {
+  // This would probably come from the server
+  const formDefaultValue: MyForm = {
+    name: 'John',
+  };
+
+  const initialState = {
+    isValid: true,
+    validate: async () => true,
+    getData: () => formDefaultValue,
+  };
```

Comment:
🤔  I get confused by this pattern. I think what trips me up is that it _looks_ like a source of truth. I read this "validate" function, which looks like a no-op and I wonder why it's defined this way. As I reason through the code, I _think_ that it must be overwritten by the true validate function as soon as the form changes... is that right?

Reply:
Yes, that's correct. As the parent component does not know anything about the form until the form calls an `onChange`, we initially have the `validate and `getData` returning the initial state.

We use this pattern when, for example, the form is in a multi step wizard and the button to go "next" is outside the scope of the component where the form is declared.

Reply:
Got it! It might be worth including this explanation in the docs in case anyone else is confused like I was.

Reply:
Good idea, I added it as a comment in the code.